android/app/src/main/java/org/fossasia/openevent/core/schedule/DayScheduleFragmentViewModel.java

```diff
@@ -0,0 +1,80 @@
+package org.fossasia.openevent.core.schedule;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.common.arch.FilterableRealmLiveData;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.repository.RealmDataRepository;
+import org.fossasia.openevent.common.utils.SortOrder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import io.reactivex.functions.Predicate;
+import io.realm.RealmResults;
+import io.realm.Sort;
+
+public class DayScheduleFragmentViewModel extends ViewModel {
+
+    private RealmDataRepository realmRepo;
+    private RealmResults<Session> sessions;
+    private List<String> tracks = new ArrayList<>();
+    private FilterableRealmLiveData<Session> filterableSessions;
+    private String searchText = "";
+
+    public DayScheduleFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public String getSearchText() { return searchText; }
+
+    public LiveData<List<Session>> getSessionsByDate(String date, String searchText) {
+        // Get entirely new data from realm
+        sessions = loadSessions(date);
+        // Pass an empty list for selectedTracks
```

Comment:
You can actually remove these comments. It's pretty clear what we're doing here. Also the method is extremely short, hence it's very easy to understand what we're doing anyways.